### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by bhandeland**

### CODE REVIEW

**Comment by antelle** [comment]
Hi!
* please branch from develop, now it's hard to follow because there are a lot of unrelated changes
* please provide some motivation
* how do these protocols work and what are they supposed to do?

**Comment by antelle** [comment]
(closed by mistake)

**Comment by bhandeland** [comment]
Thanks Antelle.  Let me see if I can do a bit better job at following process and explaining this.
I might dig a bit deeper and see if I can make this a bit less of a hack overall.

**Comment by antelle** [comment]
Probably it will be easier to open a new PR
